sympy/physics/continuum_mechanics/beam.py

```diff
@@ -170,7 +179,21 @@ def second_moment(self):
 
     @second_moment.setter
     def second_moment(self, i):
-        self._second_moment = sympify(i)
+        if isinstance(i, GeometryEntity):
+            self.cross_section = i
+            self._second_moment = i.second_moment_of_area()[0]
+        else:
+            self.cross_section = None
+            self._second_moment = sympify(i)
+
+    @property
+    def cross_section(self):
+        """Cross_section of the Beam """
+        return self._cross_section
+
+    @cross_section.setter
+    def cross_section(self, s):
+        self._cross_section = s
```

Comment:
Should this not update second_moment?

Reply:
It is being updated in the second moment setter

Reply:
So if I create a Beam with a particular second_moment and then change the geometry with `beam.cross_section = new_shape` will the `_second_moment` attribute be updated?